### ISSUE TITLE
MaxWorkingSet has wrong value when setting container memory limit

### ISSUE THREAD

**Issue by glucaci** [dotnet/runtime#81372]
### Description

When trying to use the `Process.GetCurrentProcess().MaxWorkingSet` on a linux host, it's using underlining the `cgroup` V1 or V2.
In [code](https://github.com/dotnet/runtime/blob/d099f075e45d2aa6007a22b71b45a08758559f80/src/libraries/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs#L115) and also on the [Kubernetes docs](https://kubernetes.io/docs/concepts/architecture/cgroups/#check-cgroup-version) it's only specify `cgroup2fs` and `tmpfs`.

When I did a test on our AKS cluster I receive as `DriverFormat` the `sharefs` which make than the `MaxWorkingSet` will be always `long.MaxValue`.

Trying to debug it locally in a normal container on Docker it's also `sharefs`.

Following the Kubernetes docs to identify the cgroup version it's `tmpfs`.

![image](https://user-images.githubusercontent.com/13761704/215560892-3f85fb94-68f9-432b-8ff3-3dbc327d18a3.png)

### Reproduction Steps

Create new console app, add docker support for linux, add docker run args `--memory=1g`  and use the following code:

```csharp
internal class Program
{
   private const string SysFsCgroupFileSystemPath = "/sys/fs/cgroup";

   static void Main(string[] args)
   {
      var driveInfo = new DriveInfo(SysFsCgroupFileSystemPath);
      Console.WriteLine($"DriverFormat {driveInfo.DriveFormat}");
      Console.WriteLine($"MaxWorkingSet: {Process.GetCurrentProcess().MaxWorkingSet}");
      Console.ReadLine();
   }
}
```

### Expected behavior

`MaxWorkingSet` should be equal with the output from `cat /sys/fs/cgroup/memory/memory.limit_in_bytes`

### Actual behavior

`MaxWorkingSet` is falling back to `long.MaxValue` because `sharefs` is not supported. [Interop.cgroups.cs](https://github.com/dotnet/runtime/blob/d099f075e45d2aa6007a22b71b45a08758559f80/src/libraries/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs#L115)

### Configuration

.NET 7

AKS with Kubernetes v1.24.6
Docker Desktop 4.15.0

base image used for demo application `mcr.microsoft.com/dotnet/runtime:7.0`

**Comment by glucaci** [comment]
Only by mapping the `sharefs` also to cgroup V1
```csharp
return new DriveInfo(SysFsCgroupFileSystemPath).DriveFormat switch
{
   "cgroup2fs" => CGroupVersion.CGroup2,
   "tmpfs" or "sharefs" => CGroupVersion.CGroup1,
   _ => CGroupVersion.None,
};
```
seams everything to work as expected.
```
DriveFormat: sharefs
MaxWorkingSet: 1073741824
```

**Comment by glucaci** [comment]
There are any news?

**Comment by janvorli** [comment]
This should have been fixed by #99508 in .NET 9 Preview 6.
